### PR TITLE
#706 Grid<> should have extra fields: cellWidth and cellHeight of type int

### LINKED ISSUE #706: Grid<> should have extra fields: cellWidth and cellHeight of type int
Ctor with + ctor without

### PR DESCRIPTION
**Pull request by AahzBrut**
add properties cellWidth and cellHeight to Grid<>;
add new constructor which takes cellWidth and cellHeight along with other parameters;
add tests to check that new constructor and getters are working as intended;

closes #706

### CODE REVIEW

File: fxgl-ai/src/test/java/com/almasb/fxgl/pathfinding/GridTest.java [hunk 1]
```diff
@@ -73,7 +91,7 @@ public void testGetRandomCell() {
     @Test
     public void testRandomCell() {
         for (int i = 0; i < 50; i++) {
-            Optional<MockCell> c = grid.getRandomCell(cell -> cell.getX() < 3 && cell.getY() > 15);
+            Optional<MockCell> c = grid.getRandomCell(cell -> cell.getX() < 3 && cell.getY() > CELL_HEIGHT);
```
Comment (AlmasB) [inline fxgl-ai/src/test/java/com/almasb/fxgl/pathfinding/GridTest.java:94]: Please revert the change to this line. `15` here is not `CELL_HEIGHT`, even though the value is the same. Otherwise, a change of `CELL_HEIGHT` may unintentionally break this test.

Comment (AahzBrut) [reply fxgl-ai/src/test/java/com/almasb/fxgl/pathfinding/GridTest.java]: Oops, sorry, thats was unintended, fixed
